src/modules/notifications/dtos/create-notification-attachment.dto.ts

```diff
@@ -0,0 +1,10 @@
+import { IsNotEmpty } from 'class-validator';
+import { Stream } from 'stream';
+
+export class CreateNotificationAttachmentDto {
```

Comment:
Do we really need a custom dto for adding just two defined properties? Wondering why we needed to create custom decorator for this as I just see it as an extra step as no custom logic is added in the decorator. We can just do nested check if needed like here.
https://github.com/OsmosysSoftware/osmo-notify/blob/main/src/modules/notifications/dtos/providers/wa360Dialog-data.dto.ts

Reply:
We needed custom logic like checking for channel type and using different dtos based on it in case of data field.

https://github.com/OsmosysSoftware/osmo-notify/blob/63e147a5eeb421296805a796389c6010acfb6bce/src/common/decorators/is-data-valid.decorator.ts#L32

Reply:
No, it is not necessary to have this custom decorator for validating the attachments. 

Planning to remove this file and switch to the example you provided, but I am facing an issue.

On setting validators like follows in smtp/mailgun dto files,
```ts
  @IsOptional()
  @ValidateNested({ each: true })
  @Type(() => CreateNotificationAttachmentDto)
  attachments: CreateNotificationAttachmentDto[];
```
and performing the necessary imports, the validation for attachment completely stops working. Passing invalid values like empty filename still processes the notification. I am currently trying to find the reason for this, and fix it to make the required commit.

Reply:
Cant say what the issue is. Make sure your code is proper like below. You can let me know what the behaviour is and give me branch and example to check.

```js
import { IsNotEmpty, IsOptional, ValidateNested } from 'class-validator';
import { Type } from 'class-transformer';

export class CreateNotificationAttachmentDto {
  @IsNotEmpty()
  filename: string;

  @IsNotEmpty()
  content: string | Buffer | Stream;
}

export class CreateNotificationDto {
  @IsOptional()
  @ValidateNested({ each: true })
  @Type(() => CreateNotificationAttachmentDto)
  attachments: CreateNotificationAttachmentDto[];
}

```